### PR TITLE
Fix Spotify playlist handling

### PR DESCRIPTION
**Pull request by OtterBoops**
Added bonus of "borrowing" Nana's migration of SpotifyPlaylistSourceManager to Kotlin.

### CODE REVIEW

File: FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            return null
+        }
+
+    }
+
+    override fun isTrackEncodable(track: AudioTrack): Boolean {
+        return false
+    }
+
+    @Throws(IOException::class)
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt]: Remove these. They are not useful

File: FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            //conclusion: there's room for improvement to this whole method
+        } catch (e: TrackSearcher.SearchingException) {
+            //youtube & soundcloud not available
+            return null
+        }
+
+    }
+
+    override fun isTrackEncodable(track: AudioTrack): Boolean {
+        return false
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt]: Functions like these can be simplified to one line

File: FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        try {
+            return spotifyAPIWrapper.getPlaylistDataBlocking(spotifyListId)
+        } catch (e: Exception) {
+            log.warn("Could not retrieve playlist $spotifyListId", e)
+            throw FriendlyException("Couldn't load playlist. Either Spotify is down or the playlist does not exist.", FriendlyException.Severity.COMMON, e)
+        }
+
+    }
+
+    companion object {
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/audio/source/SpotifyPlaylistSourceManager.kt]: Should be moved to the top

**Comment by napstr** [comment]
For better reviewability, it helps separating tasks like reformatting a file or converting it to kotlin from changes to the logic into individual commits.

**Comment by Frederikam** [comment]
@napstr yeah exactly.

I trust that Otter knows what he is doing with this one. We'll publicly shame him if this breaks in production :eyes:

**Comment by OtterBoops** [comment]
> For better reviewability, it helps separating tasks like reformatting a file or converting it to kotlin from changes to the logic into individual commits.

Great point. I'll keep it into consideration ..



> @napstr yeah exactly.
> 
> I trust that Otter knows what he is doing with this one. We'll publicly shame him if this breaks in production 👀

.. next time I break patron.


Jokes aside, I agree and will do it from now on.

**Pull request merged**